src/charm.py

```diff
@@ -303,12 +303,11 @@ def update(self, event):
         charm_function = self.config["charm-function"].value
         logger.info("configuring ranger %s", charm_function)
 
-        if charm_function == "admin":
-            self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
-            command, context = self._configure_ranger_admin(container)
-
         if charm_function == "usersync":
             command, context = self._configure_ranger_usersync(container)
+        else:
```

Comment:
suggestion (non-blocking): It might be better to explicitly check the function instead of putting this under `else` unless for all cases besides `usersync` you want to expose the port.

What happens if we add new `charm_function` types?

Reply:
This is what I had previously but the linter was not happy with having `command` and `context` potentially undefined. If we added a new `charm-function` we'd need to add another if statement here. This would be required anyway to configure the added plugin and specify the correct command to start the service. Additionally we have only these 2 values allowed, so this will not be a problem with the current setup. I can add a comment as a reminder to update here for any new charm-function type.
```
class FunctionType(str, Enum):
    """Enum for the `charm-function` field."""

    admin = "admin"
    usersync = "usersync"
    ```

Reply:
The assert was also not working with the linter, but added something similar.

Reply:
I don't want to hold the PR for this reason, so I will make another suggestion and please handle it however you like.

I like `assert False` for this because it clearly signals the intent that it is not a natural part of the code execution. If Python was a compiled language, we could make this check in the compile time. Instead we are enforcing it ourselves here.

So, how about something like this:
```
command, context = None, None
if charm_function == "usersync":
    command, context = self._configure_ranger_usersync(container)
elif charm_function == "admin":
    self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
    command, context = self._configure_ranger_admin(container)
else:
    assert False, "Programming error, please add your 'charm_function' to the logic."
```

I think this would make the linter happy.

Reply:
^ this is what I had in, this commit: https://github.com/canonical/ranger-k8s-operator/pull/28/commits/b814cb20f7061215a805d613c5c3d47bcad5e362, failing linter. But I agree with the sentiment of showing it's not a natural part of the code. An alternative would be to disable this particular linter code but this could have implications elsewhere

Reply:
I ended up setting the status to blocked in this case as suggested by Ali.